Mathlib/Algebra/Polynomial/Eval/Coeff.lean

```diff
@@ -223,4 +229,17 @@ end Map
 
 end CommSemiring
 
+section DivisionRing
+
+variable [DivisionRing R] [Nontrivial S] [Semiring S]
+variable {f : R →+* S} {p : R[X]}
+
+theorem map_eq_zero_iff_from_divisionRing : p.map f = 0 ↔ p = 0 :=
+  Polynomial.map_eq_zero_iff <| RingHom.injective_from_divisionRing f
+
+theorem map_ne_zero_iff_from_divisionRing : p.map f ≠ 0 ↔ p ≠ 0 :=
+  Polynomial.map_ne_zero_iff <| RingHom.injective_from_divisionRing f
```

Comment:
Please generalise to the conditions of [RingHom.injective](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/SimpleRing/Basic.html#RingHom.injective).